src/test/kotlin/uk/gov/dluhc/printapi/database/repository/PrintDetailsRepositoryIntegrationTest.kt

```diff
@@ -0,0 +1,162 @@
+                { printDetailsRepository.get(id) },
+                PrintDetailsNotFoundException::class.java
+            )
+
+            // Then
+            assertThat(ex).isNotNull.hasMessage("Print details not found for id: $id")
+        }
+    }
+
+    @Nested
+    inner class GetByRequestId {
+        @Test
+        fun `should return print details given existing print details with requestId`() {
+            // Given
+            val expected = buildPrintDetails()
+            val other = buildPrintDetails()
+
+            printDetailsRepository.save(expected)
+            printDetailsRepository.save(other)
+
+            // When
+            val actual = printDetailsRepository.getByRequestId(expected.requestId!!)
+
+            // Then
+            assertThat(actual).usingRecursiveComparison().isEqualTo(expected)
+        }
+
+        @Test
+        fun `should throw exception given non-existing print details for requestId`() {
+            // Given
+            val requestId = aValidRequestId()
+
+            // When
+            val ex = Assertions.catchThrowableOfType(
+                { printDetailsRepository.getByRequestId(requestId) },
+                PrintDetailsNotFoundException::class.java
+            )
+
+            // Then
+            assertThat(ex).isNotNull.hasMessage("Print details not found for requestId: $requestId")
+        }
+    }
+
+    @Nested
+    inner class GetAllByStatusAndBatchId {
+        @Test
+        fun `should get print details matching the status and batchID`() {
+            // Given
+            val details = buildPrintDetails()
+            printDetailsRepository.save(details)
+
+            // When
+            val results = printDetailsRepository.getAllByStatusAndBatchId(details.status, details.batchId!!)
+
+            // Then
+            assertThat(results).containsExactly(details)
+        }
+
+        @Test
+        fun `should find no matching print details given different status`() {
+            // Given
+            val differentStatus = Status.SENT_TO_PRINT_PROVIDER
+            val details = buildPrintDetails(status = Status.PENDING_ASSIGNMENT_TO_BATCH)
+            printDetailsRepository.save(details)
+
+            // When
+            val results = printDetailsRepository.getAllByStatusAndBatchId(differentStatus, details.batchId!!)
+
+            // Then
+            assertThat(results).isEmpty()
+        }
+
+        @Test
+        fun `should find no matching print details given different batchId`() {
+            // Given
+            val differentBatchId = UUID.randomUUID().toString()
+            val details = buildPrintDetails()
+            printDetailsRepository.save(details)
+
+            // When
+            val results = printDetailsRepository.getAllByStatusAndBatchId(details.status, differentBatchId)
+
+            // Then
+            assertThat(results).isEmpty()
+        }
+
+        @Test
+        fun `should get all print details by status`() {
```

Comment:
I was sure I'd seen all these tests before; and then I realized you'd renamed the test class as well as adding your new tests, so it appears in the diff as a new file 😁